src/config.js

```diff
@@ -0,0 +1,56 @@
+/* @flow */
+import requireUncached from 'require-uncached';
+
+import {createLogger} from './util/logger';
+import {UsageError} from './errors';
+
+const log = createLogger(__filename);
+
+type ApplyConfigToArgvParams = {|
+  argv: Object,
+  configObject: Object,
+  defaultValues: Object,
+|};
+
+export function applyConfigToArgv({
+  argv,
+  configObject,
+  defaultValues = {},
+}: ApplyConfigToArgvParams): Object {
+  const newArgv = {...argv};
+  for (const option in configObject) {
+    const wasValueSetOnCLI = typeof(newArgv[option]) !== 'undefined' &&
+      (argv[option] !== defaultValues[option]);
```

Comment:
For consistency, use either `argv` or `newArgv` in this entire statement. Either one is fine. Since you're only reading the old values, I think `argv` would make a bit more sense.